contracts/Comet.sol

```diff
@@ -882,23 +882,26 @@ contract Comet is CometMainInterface {
     /**
      * @notice ERC20 transfer an amount of base token to dst
      * @param dst The recipient address
-     * @param amount The quantity to transfer
+     * @param amount The quantity to transfer (principal)
      * @return true
      */
     function transfer(address dst, uint amount) override external returns (bool) {
-        transferInternal(msg.sender, msg.sender, dst, baseToken, amount);
-        return true;
+        return transferFrom(msg.sender, dst, amount);
     }
 
     /**
      * @notice ERC20 transfer an amount of base token from src to dst, if allowed
      * @param src The sender address
      * @param dst The recipient address
-     * @param amount The quantity to transfer
+     * @param amount The quantity to transfer (principal)
      * @return true
      */
-    function transferFrom(address src, address dst, uint amount) override external returns (bool) {
-        transferInternal(msg.sender, src, dst, baseToken, amount);
+    function transferFrom(address src, address dst, uint amount) override public returns (bool) {
+        if (isTransferPaused()) revert Paused();
+        if (!hasPermission(src, msg.sender)) revert Unauthorized();
+        if (src == dst) revert NoSelfTransfer();
+
+        transferBase(src, dst, safe104(amount));
```

Comment:
We need to convert `amount` to present value before using it as an argument here.

Reply:
Hm weird, I think I did it this way to specifically handle the `amount` being different, but its clearly wrong